src/sensors/utils.rs

```diff
@@ -143,6 +156,131 @@ impl ProcessTracker {
+        let process = result.next().unwrap();
+        let mut description = HashMap::new();
+        if let Some(p) = process.get(0) {
+            if let Ok(cgroups) = p.process.cgroups() {
+                let mut found = false;
+                for cg in &cgroups {
+                    if found {
+                        break;
+                    }
+                    // docker
+                    if self.regex_cgroup_docker.is_match(&cg.pathname) {
+                        description
+                            .insert(String::from("container_scheduler"), String::from("docker"));
+                        let container_id = cg.pathname.split('/').last().unwrap();
+                        description
+                            .insert(String::from("container_id"), String::from(container_id));
+                        if let Some(container) = containers.iter().find(|x| x.Id == container_id) {
+                            let mut names = String::from("");
+                            for n in &container.Names {
+                                names.push_str(&n.trim().replace("/", ""));
+                            }
+                            description.insert(String::from("container_names"), names);
+                            description.insert(
+                                String::from("container_docker_version"),
+                                docker_version.clone(),
+                            );
+                            if let Some(labels) = &container.Labels {
+                                for (k, v) in labels {
+                                    let escape_list = ["-", ".", ":", " "];
+                                    let mut key = k.clone();
+                                    for e in escape_list.iter() {
+                                        key = key.replace(e, "_");
+                                    }
+                                    description
+                                        .insert(format!("container_label_{}", key), v.to_string());
+                                }
+                            }
+                        }
+                        found = true;
+                    } else if self.regex_cgroup_kubernetes.is_match(&cg.pathname) {
+                        warn!("Kubernetes context !");
+                        // kubernetes
+                        description.insert(
+                            String::from("container_scheduler"),
+                            String::from("kubernetes"),
+                        );
+                        let container_id = cg
+                            .pathname
+                            .split('/')
+                            .last()
+                            .unwrap()
+                            .strip_prefix("docker-")
+                            .unwrap()
+                            .strip_suffix(".scope")
+                            .unwrap();
```

Comment:
You're right, it seems highly different from one setup to another. I think having a function is a good idea.

Reply:
I think it may be because I'm using the cgroupfs driver. The recommended driver is systemd but my cluster is a temp one on equinix metal and I didn't configure it 🤦‍♂️ 

Next time I'll use systemd and see if that changes things.

https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver/

Reply:
I've tried to make this part not mandatory. Could you try in your environment the latest version of the code ?

thanks 🙏🏽

Reply:
Thanks for the changes. I've retested with the systemd and cgroupfs drivers and both were fine.